sphinx_gallery/notebook.py

```diff
@@ -142,36 +145,70 @@ def add_code_cell(work_notebook, code):
     work_notebook["cells"].append(code_cell)
 
 
-def add_markdown_cell(work_notebook, text):
+def add_markdown_cell(work_notebook, text, pandoc=False):
     """Add a markdown cell to the notebook
 
     Parameters
     ----------
     code : str
         Cell content
     """
-    markdown_cell = {
-        "cell_type": "markdown",
-        "metadata": {},
-        "source": [rst2md(text)]
-    }
+    if pandoc:
+        markdown_cell = {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [text]
+        }
+    else:
+        markdown_cell = {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [rst2md(text)]
+        }
     work_notebook["cells"].append(markdown_cell)
 
 
-def fill_notebook(work_notebook, script_blocks):
+def fill_notebook(work_notebook, script_blocks, gallery_conf):
     """Writes the Jupyter notebook cells
 
+    If available, uses pypandoc to convert rst to markdown.
+
     Parameters
     ----------
     script_blocks : list
         Each list element should be a tuple of (label, content, lineno).
     """
+    pandoc = False
+    pandoc_kwargs = {}
+    if gallery_conf["pypandoc"] or isinstance(gallery_conf["pypandoc"], dict):
+        try:
+            import pypandoc  # noqa
+            # Import error raised only when function called
+            logger.info("pandoc version: %s"
+                        % (pypandoc.get_pandoc_version(),))
+        except (ImportError, OSError) as e:
+            logger.warning("'pypandoc' not available. Using Sphinx-Gallery to "
+                           "convert rst text blocks to markdown for .ipynb "
+                           "files.")
+            if e:
```

Comment:
Is this conditional necessary? Why not just always output it with the line above?

Reply:
Not familiar with python errors and wasn't sure if there would always be a message. Thanks!